server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1344,6 +1344,12 @@ RepositoryData repositoryData() {
 
     @Override
     public void getRepositoryData(ActionListener<RepositoryData> listener) {
+        // RepositoryData is the responsibility of the elected master: we shouldn't be loading it on other nodes as we don't have good
+        // consistency guarantees there, but electedness is too ephemeral to assert. We can say for sure that this node should be
+        // master-eligible, which is almost as strong since all other snapshot-related activity happens on data nodes whether they be
+        // master-eligible or not.
+        assert clusterService.localNode().isMasterNode() : "should only load repository data on master nodes";
```

Comment:
I thought about adding it like this, but I'm a little afraid we could see some tripped assertions in tests here. It's conceivable (albeit very improbable) that we failed over to another master just before calling this method isn't it?  (I could definitely see `SnapshotResiliencyTests` trip this pretty easily)

Reply:
That's why we assert we're master-eligible and not the elected master.

The code comment echoes what you said, and justifies why this is almost as strong as the real thing.